src/shared/components/oncoprint/OncoprintUtils.ts

```diff
@@ -166,6 +182,86 @@ export function getHeatmapTrackRuleSetParams(molecularAlterationType: string) {
     };
 }
 
+function getTreatmentTrackRuleSetParams(trackSpec: IHeatmapTrackSpec):RuleSetParams {
+
+    let value_range:[number, number];
+    let legend_label:string;
+    let colors:[number, number, number, number][];
+    let value_stop_points:number[];
+    let category_to_color:{[d:string]: string} | undefined;
+
+    // - Legends for treatments can be configured in two ways:
+    // 1. Larger values are `better` and appear at the right side of the legend (a.k.a. ASC sort order)
+    // 2. Smaller values are `better` and appeat at the right side of the legend (a.k.a. DESC sort order)
+    // - The pivot threshold denotes the compound concentration that is the arbitrary boundary between effective (in red)
+    // and ineffective (in blue) concentrations. Blue and red gradient to black color at the pivotThreshold value.
+    // - The most extreme value in the legend is should be the largest value in the current track group. It is:
+    // 1. Passed in along side other track specs (if possible), or...
+    // 2. Evalueated from the datapoints in the current heatmap track
+    // - When the most extreme value does not reach the pivotThreshold an arbitrary most extreme value is 
+    // calculated for representation purpouses (10*pivotThreshold value)
+    
+    legend_label = `${trackSpec.molecularProfileName}`;
+    const dataPoints = trackSpec.data;
+    const pivotThreshold = trackSpec.pivotThreshold!;
+    const sortOrder = trackSpec.sortOrder!;
+    
+    const colorWorse  = [0,0,255,1] as [number, number, number, number];
+    const colorPivot  = [0,0,0,1] as [number, number, number, number];
+    const colorBetter = [255,0,0,1] as [number, number, number, number];
+    const categoryColorOptions = [ 'rgba(230,159,0,1)', 'rgba(0,158,115,1)', 'rgba(204,121,167,1)', 'rgba(240,228,66,1)', 'rgba(86,180,233,1)' ];
+
+    const maxValue:number = trackSpec.maxProfileValue || _(dataPoints as ITreatmentHeatmapTrackDatum[]).filter((d:ITreatmentHeatmapTrackDatum) => !d.category).map('profile_data').max()!;
```

Comment:
would you ever get here and not have it already calculated?

Reply:
Eh no, that never happens in the current code. I have removed this construct since I guess that is what you would suggest if I answer this.